mmv1/products/datastream/api.yaml

```diff
@@ -0,0 +1,280 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Datastream
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://datastream.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Datastream API
+    url: https://console.cloud.google.com/apis/library/datastream.googleapis.com
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'ConnectionProfile'
+    base_url: "projects/{{project}}/locations/{{location}}/connectionProfiles"
+    create_url: "projects/{{project}}/locations/{{location}}/connectionProfiles?connectionProfileId={{connection_profile_id}}"
+    self_link: "projects/{{project}}/locations/{{location}}/connectionProfiles/{{connection_profile_id}}"
+    update_verb: :PATCH
+    update_mask: true
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/datastream/docs/create-connection-profiles'
+      api: 'https://cloud.google.com/datastream/docs/reference/rest/v1/projects.locations.connectionProfiles'
+    description: |
+      A set of reusable connection configurations to be used as a source or destination for a stream.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
```

Comment:
In the absence of a right way - I'd rather leave it called connection_profile_id because that matches the API field. would it make sense to just remove support for the `name` field since it duplicates `id`?